Add parameters for BatchNorm

minty99 opened the pull request:
This is PR for issue #65. With this commit, FL BatchNorm can receive runningMean and runningVar additionally. This is useful when loading weights from other frameworks, etc.

minty99 commented:
@jacobkahn @vineelpratap Any updates for this?

minty99 commented:
@jacobkahn @vineelpratap @jcai1 ping!

vineelpratap commented on flashlight/nn/modules/BatchNorm.cpp:
```diff
@@ -59,27 +59,32 @@ Variable BatchNorm::forward(const Variable& input) {
-    auto bs = constant(0.0, featSize_, af::dtype::f32, true);
-    params_ = {wt, bs};
-  }
+  params_ = {wt, bs, runningMean_, runningVar_};
```
I have looked at PyTorch, TF implementations. This doesn't seem to be the standard. 

Can we just have functions called `setRunningMean`, `getRunningMean`, `setRunningVar`, `getRunningVar`. I guess that would work for you ?

minty99 replied:
Yes, it will be good. I will revise this to add functions you said soon.

jacobkahn replied:
@minty99 any update here? Would be nice to rebase and land these.

minty99 replied:
@jacobkahn I'm sorry but I was busy. I will work on this again soon.

minty99 commented:
@jacobkahn I updated the commits including your review, and rebased it on current master. However I didn't test this - I don't have test environment now. Could you review this?

jacobkahn commented:
@minty99 — the changes look fine, but we'd want to add some tests before merging.

minty99 commented:
@jacobkahn Do you have some guideline for adding tests? I think I should edit `tests/nn/ModuleTest.cpp`.

jacobkahn commented:
@minty99 — that's exactly right; those tests should go in `ModuleTest.cpp`.
